tiledb/sm/rest/curl.cc

```diff
@@ -180,22 +180,66 @@ static int buffer_list_seek_callback(
   return CURL_SEEKFUNC_FAIL;
 }
 
+/**
+ * This callback function gets called by libcurl as soon as a header has been
+ * received. libcurl buffers headers and delivers only "full" headers, one by
+ * one, to this callback. This callback should return the number of bytes
+ * actually taken care of if that number differs from the number passed to
+ * your callback function, it signals an error condition to the library
+ *
+ * @param res_data points to the delivered data
+ * @param tSize the size of that data is size multiplied with nmemb
+ * @param tCount number of bytes actually taken care of
+ * @param userdata the userdata argument is set with CURLOPT_HEADERDATA
+ * @return
+ */
+
+size_t OnReceiveData(
+    void* res_data, size_t size, size_t count, void* userdata) {
+  size_t length = size * count;
+  size_t index = 0;
+  auto* buffer = static_cast<unsigned char*>(res_data);
+
+  while (index < length) {
+    unsigned char* temp = buffer + index;
+    if ((temp[0] == '\r') || (temp[0] == '\n'))
+      break;
+    index++;
+  }
+
+  std::string str(buffer, buffer + index);
```

Comment:
use `index` as the second argument. i don't know what constructor this is intended to call:
http://www.cplusplus.com/reference/string/string/string/

my best guess is that this calls constructor (7) which needlessly constructors InputIterators